app/assets/src/components/views/bulk_download/BulkDownloadModal.jsx

```diff
@@ -118,102 +98,65 @@ class BulkDownloadModal extends React.Component {
-              displayName,
-              prevState.selectedFieldsDisplay
-            )
-          : unset([downloadType, fieldType], prevState.selectedFieldsDisplay);
-
-      return {
-        selectedFields: newSelectedFields,
-        selectedFieldsDisplay: newSelectedFieldsDisplay,
-      };
-    });
-  };
+    const backgroundOptions = backgrounds.map(background => ({
+      text: background.name,
+      value: background.id,
+    }));
 
-  handleChooseStepContinue = () => {
-    this.setState({ currentStep: "review" });
-  };
-  handleBackClick = () => {
-    this.setState({ currentStep: "choose" });
-  };
+    return backgroundOptions;
+  }
+
+  // We use the heatmap metrics as the valid metrics for bulk downloads.
+  async fetchHeatmapMetrics() {
+    const heatmapMetrics = await getHeatmapMetrics();
 
-  renderStep = () => {
+    return heatmapMetrics;
+  }
+
+  async checkAllSamplesUploadedByCurrentUser() {
+    const { selectedSampleIds } = this.props;
+    const allSamplesUploadedByCurrentUser = await uploadedByCurrentUser(
+      Array.from(selectedSampleIds)
+    );
+
+    return allSamplesUploadedByCurrentUser;
+  }
+
+  // *** Render methods ***
+
+  render() {
+    const { open, onGenerate, onClose } = this.props;
     const {
-      currentStep,
       bulkDownloadTypes,
-      selectedDownloadTypeName,
-      selectedFields,
-      selectedFieldsDisplay,
       validSampleIds,
       invalidSampleNames,
       validationError,
+      backgroundOptions,
+      metricsOptions,
+      allSamplesUploadedByCurrentUser,
     } = this.state;
 
-    if (currentStep === "choose") {
-      return (
-        <ChooseStep
-          downloadTypes={bulkDownloadTypes}
-          selectedDownloadTypeName={selectedDownloadTypeName}
-          onSelect={this.handleSelectDownloadType}
-          selectedFields={get(selectedDownloadTypeName, selectedFields)}
-          onFieldSelect={this.handleFieldSelect}
-          onContinue={this.handleChooseStepContinue}
-          validSampleIds={validSampleIds}
-          invalidSampleNames={invalidSampleNames}
-          validationError={validationError}
-        />
-      );
-    }
-
-    if (currentStep === "review") {
-      const selectedDownload = assembleSelectedDownload(
-        selectedDownloadTypeName,
-        selectedFields,
-        selectedFieldsDisplay,
-        validSampleIds
-      );
-
-      const selectedDownloadType = find(
-        ["type", selectedDownloadTypeName],
-        bulkDownloadTypes
-      );
-
-      return (
-        <ReviewStep
-          selectedDownload={selectedDownload}
-          downloadType={selectedDownloadType}
-          onBackClick={this.handleBackClick}
-        />
-      );
-    }
-  };
-
-  render() {
-    const { open } = this.props;
-
     return (
-      <Modal narrow open={open} tall onClose={this.props.onClose}>
-        {this.renderStep()}
+      <Modal narrow open={open} tall onClose={onClose}>
+        {
+          //loading && "L O A D I N G"
```

Comment:
Do we need this comment? What does it mean?

Reply:
That should not have been there, oops.